test/jdk/javax/management/mxbean/MXBeanLoadingTest1.java

```diff
@@ -80,38 +80,9 @@ public void run(Map<String, Object> args) {
                     replaceAll(MXBeanLoadingTest1.class.getSimpleName()
                             + ".class", "");
 
-            URL[] urls = new URL[]{new URL(clsLoadPath)};
-            @SuppressWarnings("removal")
-            PrivateMLet mlet = new PrivateMLet(urls, null, false);
-            Class<?> shadowClass = mlet.loadClass(TestMXBean.class.getName());
-
-            if (shadowClass == TestMXBean.class) {
-                String message = "(ERROR) MLet got original TestMXBean, not shadow";
-                System.out.println(message);
-                throw new RuntimeException(message);
-            }
-            shadowClass = null;
-
             MBeanServer mbs = MBeanServerFactory.createMBeanServer();
-            ObjectName mletName = new ObjectName("x:type=mlet");
-            mbs.registerMBean(mlet, mletName);
-
             ObjectName testName = new ObjectName("x:type=test");
-            mbs.createMBean(Test.class.getName(), testName, mletName);
-
-            // That test fails because the MXBean instance is accessed via
-            // a delegate OpenMBean which has
-            ClassLoader testLoader = mbs.getClassLoaderFor(testName);
-
-            if (testLoader != mlet) {
-                System.out.println("MLet " + mlet);
-                String message = "(ERROR) MXBean's class loader is not MLet: "
-                        + testLoader;
-                System.out.println(message);
-                throw new RuntimeException(message);
-            }
-            testLoader = null;
-
+            ObjectInstance mb = mbs.createMBean(Test.class.getName(), testName);
```

Comment:
Same remark here - I think we need to first register an MBean which is a ClassLoader and implements PrivateClassLoader to replace the MLet in this test.

Reply:
Updated!